File: src/modules/lsp/serverConfigs/rust.ts
// rust-analyzer 的前端元数据（实际 binary 由 Rust 端 lsp_resolve_command 解析）。
import type { LspServerSpec } from "../types";

export const RUST_SPEC = {
  id: "rust-analyzer",
  language: "rust",
} as const satisfies Pick<LspServerSpec, "id" | "language">;

File: src-tauri/src/modules/lsp/servers/rust.rs
use std::path::Path;

use super::super::LspServerSpec;

pub fn spec(_cwd: &Path) -> Option<LspServerSpec> {
    which::which("rust-analyzer")
        .ok()
        .map(|path| LspServerSpec {
            id: "rust-analyzer".into(),
            language: "rust".into(),
            command: path.to_string_lossy().into_owned(),
            args: Vec::new(),
            cwd: None,
        })
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn returns_spec_or_none() {
        // 不强制 binary 存在；CI 上可能没有安装。
        let _ = spec(Path::new("/tmp"));
    }
}
